src/main/java/org/myrobotlab/service/WebGui.java

```diff
@@ -979,6 +996,22 @@ public String redirect(String url) {
 
   public void restart() {
     stop();
+    WebGui _self = this;
+
+    // done so a thread "from" webgui can restart itself
+    // similar thread within stop
+    // From a web request you cannot block on a request to stop/start self
+    new Thread() {
+      public void run() {
+        try {
+          while (nettosphere != null) {
+            Thread.sleep(500);

```

Comment:
is there a way to avoid using Thread.sleep(..)  here? if not, no big deal.

Reply:
The de-allocation of nettosphere is not in a good place..
I got a couple of prs accepted into nettosphere - but I think there is still some race condition problems